rust/noosphere/src/ffi/fs.rs

```diff
@@ -241,3 +242,36 @@ pub fn ns_sphere_file_header_values_read(
         .into_boxed_slice()
         .into()
 }
+
+#[ffi_export]
+/// Get the first header value for a file that corresponds to the given name, if any
+pub fn ns_sphere_file_header_value_first(
+    sphere_file: &NsSphereFile,
+    name: char_p::Ref<'_>,
+) -> Option<char_p::Box> {
+    sphere_file
+        .inner
+        .memo
+        .get_first_header(name.to_str())
+        .into_iter()
+        .nth(0)
+        .map(|value| value.try_into().unwrap())
+}
+
+#[ffi_export]
+/// Read all the headers associated with a file as an array of strings. Note
+/// that headers will be reduced to a single entry in cases where multiple
```

Comment:
If I understand right, we drop duplicate headers, rather than taking the HTTP approach of concatenating them with a comma?

https://www.rfc-editor.org/rfc/rfc2616.html#section-4.2

> Multiple message-header fields with the same field-name MAY be
   present in a message if and only if the entire field-value for that
   header field is defined as a comma-separated list [i.e., #(values)].
   It MUST be possible to combine the multiple header fields into one
   "field-name: field-value" pair, without changing the semantics of the
   message, by appending each subsequent field-value to the first, each
   separated by a comma. The order in which header fields with the same
   field-name are received is therefore significant to the
   interpretation of the combined field value, and thus a proxy MUST NOT
   change the order of these field values when a message is forwarded.

FWIW, I think dropping duplicate headers is probably preferable for now. I feel we want to discourage the comma-separated value headers, since it makes assumptions about the header value format.

Reply:
Ah, nvm this only retrieves unique header KEYS. That's perfect.

Reply:
No comment on dropping dupes vs multiple headers, but the current Swift test has a header entry `"hello:noosphere"` that cannot be retrieved with the current FFI as an example; whether dupe headers or dropping dupes, the structs should reflect that

Reply:
It can be retrieved with the API that existed before these helpers were added @jsantell `ns_sphere_file_header_values_read` gives you all values for a name.

Reply:
perfect!!